app/models/lr_common_config.rb

```diff
@@ -13,6 +13,8 @@ def app_link(app_name, request)
       end
 
       path = relative_url_root.sub(%r{[^/]*\Z}, app_name)
+      path += "?lang=#{request.params[:lang]}" if request.params.key?(:lang)
+
```

Comment:
Is it worth checking that the param is a supported language? Maybe this is taken care of already elsewhere...

Reply:
Good observation. I think it should be OK. Unrecognised languages will just default to English (as the enterprising person who amended the param to `lang=pl` would have found out!)